src/components/views/userProfile/donationsTab/recurringTab/FilterMenu.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+	B,
+	Button,
+	ButtonText,
+	IconX,
+	neutralColors,
+} from '@giveth/ui-design-system';
+import styled from 'styled-components';
+import { useIntl } from 'react-intl';
+import { forwardRef } from 'react';
+import { useRouter } from 'next/router';
+import { mediaQueries } from '@/lib/constants/constants';
+import { zIndex } from '@/lib/constants/constants';
+import CheckBox from '@/components/Checkbox';
+import { FlexCenter } from '@/components/styled-components/Flex';
+import config from '@/configuration';
+import { ISuperToken } from '@/types/superFluid';
+import { PinkyColoredNumber } from '@/components/styled-components/PinkyColoredNumber';
+
+interface IFilterMenuProps {
+	handleClose: (e?: any) => void;
+	isOpen?: boolean;
+}
+
+export const FilterMenu = forwardRef<HTMLDivElement, IFilterMenuProps>(
+	({ handleClose, isOpen }, ref) => {
+		const { formatMessage } = useIntl();
+		const count = 0;
+		const router = useRouter();
+
+		const handleSelectFilter = (e: boolean, filter: ISuperToken) => {
+			handleClose();
+		};
+
+		const clearFilters = () => {
+			const updatedQuery = {
+				...router.query,
+			};
+			delete updatedQuery.filter;
+			delete updatedQuery.campaign;
+			router.push({
+				pathname: router.pathname,
+				query: updatedQuery,
+			});
+		};
+
+		return (
+			<MenuContainer className={isOpen ? 'fadeIn' : 'fadeOut'} ref={ref}>
+				<Header>
+					<CloseContainer onClick={handleClose}>
+						<IconX size={24} />
+					</CloseContainer>
+					<FlexCenter gap='8px'>
+						<ButtonText size='medium'>
+							{formatMessage({ id: 'label.filters' })}
+						</ButtonText>
+						{count !== 0 && (
+							<PinkyColoredNumber size='medium'>
+								{count}
+							</PinkyColoredNumber>
+						)}
+					</FlexCenter>
+				</Header>
+				<Section>
+					<B>{formatMessage({ id: 'label.tokens' })}</B>
+					{config.OPTIMISM_CONFIG.SUPER_FLUID_TOKENS.map(token => (
+						<FeatureItem key={token.id}>
+							<CheckBox
+								label={token.name}
+								onChange={e => {
+									handleSelectFilter(e, token);
+								}}
+								checked={
+									false
```

Comment:
please confirm if these needs to be dynamic instead of false always

Reply:
@jainkrati I will handle it when backend is ready